*: Remove serial connection option

mxinden opened the pull request:
With previous commits the serial connection option was left untested and
non-functional. Instead of leaving the logic around to be fixed in the
future, this commit removes the logic. Thereby TCP is the only option.
Once serial is needed, this commit can be reverted.

RichiH commented:
LGTM, but two points:

1) Please point the commit ID out in the README
2) The config should still take a Modbus TCP as an option to make explicit what protocol is being used. It's fine to default to Modbus TCP if no option is given

mxinden committed:
*: Remove serial connection option

With previous commits the serial connection option was left untested and
non-functional. Instead of leaving the logic around to be fixed in the
future, this commit removes the logic. Thereby TCP is the only option.
Once serial is needed, this commit can be reverted.

mxinden committed:
README.md: Reference commit dropping support for serial modbus

mxinden commented:
@RichiH sure. Resurrected the `tcp/ip` config property and referenced commit hash in README.md.
